frontend/app-development/features/dataModelling/DataModelling.test.tsx

```diff
@@ -53,63 +53,45 @@ const render = (queryClient: QueryClient = createQueryClientMock()) => {
-    render(queryClient);
-    const dialogHeader = screen.getByRole('heading', { name: textMock('schema_editor.info_dialog_title') });
-    expect(dialogHeader).toBeInTheDocument();
-  });
-
-  it('should display no data-models message when schema is undefined and loadState is loaded', async () => {
-    localStorage.removeItem(LOCAL_STORAGE_KEY);
-    const queryClient = createQueryClientMock();
-    queryClient.setQueryData([QueryKey.DatamodelsMetadata, org, app], [jsonMetadata1Mock]);
-    render(queryClient);
-    const dialogHeader = screen.getByRole('heading', { name: textMock('schema_editor.info_dialog_title') });
-    expect(dialogHeader).toBeInTheDocument();
-  });
-
-  it('Should not show info dialog when loading the page if user has asked to not show it again', () => {
-    // make sure setting to turn off info dialog is set
-    setLocalStorageItem('hideIntroPage', true);
-    render();
-    const dialogHeader = screen.queryByRole('heading', { name: textMock('schema_editor.info_dialog_title') });
-    expect(dialogHeader).not.toBeInTheDocument();
-  });
-
-  it('Should show start dialog when no models are present and intro page is closed', () => {
-    // make sure setting to turn off info dialog is set
-    setLocalStorageItem('hideIntroPage', true);
+  it('shows start dialog when no models are present and intro page is closed', () => {
     const queryClient = createQueryClientMock();
     queryClient.setQueryData([QueryKey.DatamodelsMetadata, org, app], []);
     render(queryClient);
     const dialogHeader = screen.getByRole('heading', { name: textMock('app_data_modelling.landing_dialog_header') });
     expect(dialogHeader).toBeInTheDocument();
   });
 
-  it('Should not show start dialog when the models have not been loaded yet', () => {
-    // make sure setting to turn off info dialog is set
-    setLocalStorageItem('hideIntroPage', true);
+  it('does not show start dialog when the models have not been loaded yet', () => {
     render();
     expect(screen.getByTitle(textMock('general.loading'))).toBeInTheDocument();
     expect(screen.queryByRole('heading', { name: textMock('app_data_modelling.landing_dialog_header') })).not.toBeInTheDocument();
   });
 
-  it('Should not show start dialog when there are models present', async () => {
-    // make sure setting to turn off info dialog is set
-    setLocalStorageItem('hideIntroPage', true);
+  it('does not show start dialog when there are models present', async () => {
     getDatamodels.mockImplementation(() => Promise.resolve([jsonMetadata1Mock]));
     render();
     await waitForElementToBeRemoved(() => screen.queryByTitle(textMock('general.loading')));
     expect(screen.queryByRole('heading', { name: textMock('app_data_modelling.landing_dialog_header') })).not.toBeInTheDocument();
   });
+
+  it('shows an error message if an error occured', async () => {
+    const errorMessage = 'error-message-test';
+    jest
+    .spyOn(reactQuery, 'useQuery')
```

Comment:
What do you think about changing this mock to mock the response instead of ReactQuery itself? Something similar TextEditor.test.tsx line 19 and use Promise.Reject instead of Promise.Resolve(). 🤔 😄

Reply:
Done!